### DISCUSSION TITLE
What is the `metadata` field in `DataSchema` used for?

### DISCUSSION THREAD

**Discussion by RinChanNOWWW** [databendlabs/databend#5939] [Q&A] [answered]
https://github.com/datafuselabs/databend/blob/93621e902bd172cb4fd60fab4a90d927b6e3d2be/common/datavalues/src/data_schema.rs#L26-L31

I'm implementing comments for columns now. I find that there is a `metadata` field in `DataSchema`, and I'm wondering if I should put the comment in `metadata` or in `DataField`.

**Comment by bohutang** [comment]
Has not seen anywhere to use it, I think @youngsofun may know more about it.

**Comment by youngsofun** [comment]
@RinChanNOWWW it`s explained here https://databend.rs/doc/contributing/rfcs/new-sql-planner-framework#binder

**Reply by RinChanNOWWW** [reply]
It seems that the metadata in binder is not the same with the one in `DataSchema`. And `DataSchema.metadata` is never written.

**Reply by youngsofun** [reply]
sorry, I was busy and misread your question.

it is corrisponding to metadata of Arrow schema.  
not sure whether it will be used later

but itself looks more like a table level metadata, (not field level).

and I think we can just store column comments  in tableInfos in catelog . don`t  need to carry it in Schema.

**Comment by sundy-li** [comment, marked as answer]
Metadata will be used to convert into Arrow's schema. See:  https://github.com/jorgecarleitao/arrow2/blob/main/src/datatypes/schema.rs#L13-L29

**Reply by RinChanNOWWW** [reply]
So it's just for keeping same with Arrow?

**Reply by sundy-li** [reply]
Yes
